src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTask.java

```diff
@@ -77,9 +77,15 @@ public void start(Map<String, String> props, JestClient client) {
       if (client != null) {
         this.client = client;
       } else {
-        List<String> address = config.getList(ElasticsearchSinkConnectorConfig.CONNECTION_URL_CONFIG);
         JestClientFactory factory = new JestClientFactory();
-        factory.setHttpClientConfig(new HttpClientConfig.Builder(address).multiThreaded(true).build());
+        List<String> addresses = config.getList(ElasticsearchSinkConnectorConfig.CONNECTION_URL_CONFIG);
+        HttpClientConfig.Builder clientBuilder = new HttpClientConfig.Builder(addresses).multiThreaded(true);
+        String username = config.getString(ElasticsearchSinkConnectorConfig.CONNECTION_USERNAME);
+        if (username != null) {
+          String password = config.getString(ElasticsearchSinkConnectorConfig.CONNECTION_PASSWORD);
```

Comment:
FWIW, this patch works for us in v 3.2.1, however we had to change this line to:
`String password = config.getPassword(ElasticsearchSinkConnectorConfig.CONNECTION_PASSWORD).value()`

I'm not sure if anything has changed up stream in master that makes this update unnecessary, but thought I'd mention it.